### PR TITLE
Address comments from #6104

### PR DESCRIPTION
**Pull request by kisieland**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Addresses comments, that were provided by liggitt@ on #6104 after the PR was already merged.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/assign mwielgus
/assign liggitt

### CODE REVIEW

File: cluster-autoscaler/provisioningrequests/apis/autoscaling.x-k8s.io/v1beta1/types.go [hunk 1]
```diff
@@ -139,13 +160,16 @@ type ProvisioningRequestStatus struct {
...
 	Conditions []metav1.Condition `json:"conditions"`
 
-	// AdditionalStatus contains all other status values custom provisioning classes may require.
+	// Statuses contains all other status values custom provisioning classes may require.
 	//
 	// +optional
-	// +kubebuilder:validation:MinProperties=64
-	AdditionalStatus map[string]string `json:"additionalStatus"`
+	// +kubebuilder:validation:MaxProperties=64
+	Statuses map[string]Status `json:"statuses"`
```
Comment (liggitt) [inline cluster-autoscaler/provisioningrequests/apis/autoscaling.x-k8s.io/v1beta1/types.go:167]: `.status.statuses` stutters a bit... trying to think of a non-"status" name for this... Details, ProvisioningClassDetails, maybe?

File: cluster-autoscaler/provisioningrequests/apis/autoscaling.x-k8s.io/v1beta1/types.go [hunk 2]
```diff
@@ -112,7 +131,7 @@ type PodSet struct {
 	// template.
 	//
 	// +kubebuilder:validation:Minimum=1
-	// +kubebuilder:validation:Maximum=16384
+	// +kubebuilder:validation:Maximum=2147483647
```
[the remark this answers is not in the document]
Comment (kisieland) [reply cluster-autoscaler/provisioningrequests/apis/autoscaling.x-k8s.io/v1beta1/types.go:134]: Done

File: cluster-autoscaler/provisioningrequests/apis/autoscaling.x-k8s.io/v1beta1/types.go (hunk 1 above)
Comment (kisieland) [reply cluster-autoscaler/provisioningrequests/apis/autoscaling.x-k8s.io/v1beta1/types.go:167]: ProvisioningClassDetails it is

**Comment by kisieland** [comment]
@liggitt PTAL

**Comment by liggitt** [comment]
did a quick pass, changes lgtm

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Pull request merged**